test/metabase/test/data/materialize.clj

```diff
@@ -0,0 +1,104 @@
+(ns metabase.test.data.materialize
+  "Test extensions for the Materialize driver.  Includes logic for creating/destroying test datasets, building
+  the connection specs from environment variables, etc."
+  (:require
+   [clojure.string :as str]
+   [metabase.config :as config]
+   [metabase.driver :as driver]
+   [metabase.driver.ddl.interface :as ddl.i]
+   [metabase.test.data.interface :as tx]
+   [metabase.test.data.sql :as sql.tx]
+   [metabase.test.data.sql-jdbc.execute :as execute]
+   [metabase.test.data.sql-jdbc.load-data :as load-data]
+   [metabase.test.data.sql.ddl :as ddl]
+   [metabase.util.log :as log]))
+
+(set! *warn-on-reflection* true)
+
+(defmethod driver/database-supports? [:materialize :foreign-keys] [_driver _feature _db] (not config/is-test?))
+
+(defmethod ddl/drop-db-ddl-statements :materialize
+  [& args]
+  (apply (get-method ddl/drop-db-ddl-statements :sql-jdbc/test-extensions) args))
+
+(defmethod tx/aggregate-column-info :materialize
+  ([driver ag-type]
+   ((get-method tx/aggregate-column-info ::tx/test-extensions) driver ag-type))
+
+  ([driver ag-type field]
+   (merge
+    ((get-method tx/aggregate-column-info ::tx/test-extensions) driver ag-type field)
+    (when (= ag-type :sum)
+      {:base_type :type/BigInteger}))))
+
+(doseq [[base-type db-type] {:type/BigInteger     "BIGINT"
+                             :type/Boolean        "BOOL"
+                             :type/Date           "DATE"
+                             :type/DateTime       "TIMESTAMP"
+                             :type/Decimal        "DECIMAL"
+                             :type/Float          "FLOAT"
+                             :type/Integer        "INTEGER"
+                             :type/Text           "TEXT"
+                             :type/JSON           "JSON"
+                             :type/Time           "TIME"
+                             :type/UUID           "UUID"}]
+  (defmethod sql.tx/field-base-type->sql-type [:materialize base-type] [_ _] db-type))
```

Comment:
Aren't these inherited from the postgres driver?

Reply:
If I don't specify those, one of the tests is failing with the following error:

```clj
clojure.lang.ExceptionInfo: Failed to create :materialize 'office-checkins' test database: No test data type mapping for driver :materialize for base type :type/DateTimeWithZoneOffset; add an impl for field-base-type->sql-type.
    database-name: "office-checkins"
           driver: :materialize
       java.lang.Exception: No test data type mapping for driver :materialize for base type :type/DateTimeWithZoneOffset; add an impl for field-base-type->sql-type.
```